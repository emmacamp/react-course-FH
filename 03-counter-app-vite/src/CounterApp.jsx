import Proptypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = () => setCounter(counter + 1);
    const handleSub = () => setCounter(counter - 1);
    const handleReset = () => setCounter(value);


    return (
        <div>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={handleAdd}>
                +1
            </button>
            <button onClick={handleSub}>
                -1
            </button>
            <button aria-label='btn-reset' onClick={handleReset}>
                Reset
            </button>
        </div>
    )
}


CounterApp.propTypes = {
    value: Proptypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}