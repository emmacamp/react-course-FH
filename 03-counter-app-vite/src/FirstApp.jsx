import Proptypes from 'prop-types'

// const newMessage = { name: 'Pepe', age: 10 }

export const FirstApp = ({ title, subTitle, name }) => {
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <p>{name}</p>
        </>
    )
}


FirstApp.propTypes = {
    title: Proptypes.string.isRequired,
    subTitle: Proptypes.number.isRequired
}

FirstApp.defaultProps = {
    // title: 'Default title',
    subTitle: 'Im subtitle',
    name: 'user'
}