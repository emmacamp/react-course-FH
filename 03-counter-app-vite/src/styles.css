html,
body {
    background-color: #21232A;
    color: #fff;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button {
    background-color: #010101;
    border: none;
    border-radius: 5px;
    color: #fff;
    cursor: pointer;
    font-size: 1.3rem;
    padding: 10px 20px;
    margin: 0 5px;
}