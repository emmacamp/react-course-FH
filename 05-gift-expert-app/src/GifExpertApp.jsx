import { useState } from "react";
import { AddCategories, GiftGrid } from "./components";
// import homeSVG from './assets/aroundTheWorld.svg'

const GifExpertApp = () => {
  const [category, setCategory] = useState([]);

  const onAddCategory = (newCategory) => {
    if (category.includes(newCategory)) return;
    setCategory([newCategory, ...category])
  }

  return (
    <div className="container vh-75 justify-content-center  align-items-center">
      <div className="header p-2 justify-content-between  align-items-center">
        <h1 className="p-0 m-0">GifExpertApp</h1>
        <AddCategories addCategories={onAddCategory} stateCategories={category} />
      </div>
      <div className="container mw-100 d-flex justify-content-center  align-items-center">
        {
          category.length === 0
            ?
            <div className="d-flex flex-column w-50 h-25">
              <img className="w-100 " src={'http://perros.com/macho/doberman'} alt="Home image" />
              {/* <h1>Busca tu gif favorito</h1> */}
            </div>
            :
            category.map((ctg) => (
              <GiftGrid key={ctg} category={ctg} /> || <h1>No se encontraron datos</h1>
            ))
        }
      </div>
    </div>
  )
}



export default GifExpertApp;