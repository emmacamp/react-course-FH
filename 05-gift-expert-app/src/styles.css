:root {
    --primary-color: rgb(97, 32, 158);
    --secondary-color: rgb(234, 234, 234);
    --tertiary-color: white;
    --quaternary-color: black;
    --quinary-color: rgb(0, 0, 0, 0.03);
}

* {
    font-family: Helvetica, Arial, sans-serif;
    /* background-color: rgb(234, 234, 234); */
    padding: 2px;
}

.container {
    background-color: var(--secondary-color);
    box-shadow: 0px 3px 5px var(--quinary-color);

}



.header {
    background-color: var(--primary-color);
    border-radius: 5px;
    color: var(--tertiary-color);
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
    padding: 0 10px;

}

.header h1 {
    background-color: var(--primary-color);
    padding: 0;
}

form {
    display: flex;
    flex-direction: row-reverse;
    background: transparent;
    gap: 5px;
    height: 50px;
}

input {
    background-color: var(--tertiary-color);
    border-radius: 5px;
    border: 1px solid var(--primary-color);
    color: var(--quaternary-color);
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 400px;
}

button {
    background-color: var(--primary-color);
    border-color: rgba(0, 0, 0, 0.81);
    border-radius: 5px;
    color: white;
    font-size: 1.2rem;
    font-weight: bold;
    padding: 10px 15px;
    transition: .2s all ease-out;

}

button:hover {
    /* transition: .2s all ease-out; */
    background-color: var(--quaternary-color);
    color: rgb(168, 100, 232);
    cursor: pointer;
}

h2 {
    font-size: 1.5rem;
}

h3 {
    font-size: 3rem;
    margin-bottom: 5px;
}

.card-grid {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
}

.card {
    align-content: center;
    align-items: center;
    background-color: var(--tertiary-color);
    border-radius: 10px;
    box-shadow: 0px 3px 5px var(--quinary-color);
    display: flex;
    flex-direction: column;
    height: 0%;
    justify-content: center;
    margin-bottom: 20px;
    margin-right: 20px;
    overflow: hidden;
    width: 300px;
    height: 300px;
}

.card p {
    background-color: var(--secondary-color);
    flex: 1;
    font-size: 1.5rem;
    margin-top: 5px;
    padding: 10px;
    text-align: center;
}

.card img {
    max-width: 300px;
    max-height: 300px;
}