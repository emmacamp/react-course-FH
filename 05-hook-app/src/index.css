body {
  padding: 20px;
  background-color: black;
  color: aliceblue;
}

button {
  margin-right: 10px;
}