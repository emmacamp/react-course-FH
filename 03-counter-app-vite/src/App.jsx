import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';

export const App = () => {


    return (
        <>
            {/* <FirstApp title='Hola Pepe' /> */}
            <CounterApp value={100} />
        </>
    )
}
