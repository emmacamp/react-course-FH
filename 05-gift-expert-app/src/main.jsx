import ReactDOM from 'react-dom/client'

// bootstrap
import 'bootstrap/dist/css/bootstrap.min.css'


import './styles.css'
import GifExpertApp from './GifExpertApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp />
  </React.StrictMode>,
)
