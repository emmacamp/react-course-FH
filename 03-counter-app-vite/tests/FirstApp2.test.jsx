import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirstApp2 />', () => {
    const title = 'Hola Pepe'
    const subTitle = 'Im subtitle'


    test('Debe de hacer match con el snapshot', () => {

        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot()

    })

    test('Debe mostrar el mensaje "Hola Pepe"', () => {

        render(<FirstApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy()

        // screen.debug()
    })

    test('Debe mostrar el titulo en un h1', () => {

        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toBeTruthy();

    })

    test('Debe mostrar un subtitulo enviado por props', () => {

        render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />);

        expect(screen.getAllByText(subTitle)).toBeTruthy();
    })

})