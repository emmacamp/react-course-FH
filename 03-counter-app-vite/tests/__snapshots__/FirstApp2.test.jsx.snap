// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2 /> Debe hacer match del screenshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Pepe
  </h1>
  <h2>
    Im subtitle
  </h2>
  <h2>
    Im subtitle
  </h2>
  <p>
    user
  </p>
</div>
`;
